### PR TITLE
Fix transitive dep resolution when exclusion is present in other POM

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
#908 uncovered a bug in `coursier` which leads to transitive dep not being resolved if an `exclusion` rule for it is present in POM of a non-related dependency.

This PR adds a regression test and a fix by @rdesgroppes.

TL;TR:

1. `mockito-core` depends on `byte-buddy`
2. `androidx.arch.core:core-testing` [POM](https://dl.google.com/android/maven2/androidx/arch/core/core-testing/2.1.0/core-testing-2.1.0.pom) has `exclusion` for `<groupId>net.bytebuddy</groupId>`
3. If both are present in `maven_install` then `mockito-core` doesn't get its `byte-buddy` dependency
4. If `androidx.arch.core:core-testing` is removed from `maven_install` then `mockito-core` gets `byte-buddy` dependency
5. Adding `byte-buddy` manually to `maven_install` doesn't help
6. `mockito-core` was getting `byte-buddy` dependency with `androidx.arch.core:core-testing` in `maven_install` prior to change #908 
7. This is likely a bug in coursier due to how #908 passes classifier and type for dependency resolution

### CODE REVIEW

File: tests/custom_maven_install/regression_testing_install.json [hunk 1]
```diff
@@ -1390,16 +1426,22 @@
...
+        "jar": "5e168368fbc250af3c79aa5fef0c3467a2d64e5a7bd74005f25d8399aeb0708d"
       },
-      "version": "2.1"
+      "version": "2.6"
     },
     "org.openjfx:javafx-base": {
       "shasums": {
         "jar": "c5084a74417a89c69a0c122fae96a4b70bf619fc3d6218ea102a4047ec85ad04",
-        "linux": "2ebf6fa2cbbe1c8b4f7780e06e97beb038f644d5ecf9f15a41c5e88ee0ef9cf1"
+        "mac": "2d8052a08fd2e5d98e1d5a16d724ea5dd02102879de20a193225f57199803983"
```
Comment (artem-zinnatullin) [inline tests/custom_maven_install/regression_testing_install.json:1444]: This is unfortunate, but I don't have a Linux machine/image with all required Bazel deps to regenerate this for Linux and have matching hashes

File: WORKSPACE [hunk 2]
```diff
@@ -257,6 +257,10 @@ maven_install(
         # Any two platforms to ensure that it doesn't work _only_ under the host operating system
         "com.google.protobuf:protoc:exe:linux-x86_64:3.21.12",
         "com.google.protobuf:protoc:exe:osx-aarch_64:3.21.12",
+        # https://github.com/bazelbuild/rules_jvm_external/issues/917
+        # androidx core-testing POM has "exclusion" for "byte-buddy" but it should be downloaded as mockito-core dependency.
+        "org.mockito:mockito-core:jar:3.3.3",
```
Comment (rdesgroppes) [inline WORKSPACE:262]: Since `jar` is the default, it's usually omitted.

File: WORKSPACE [hunk 3]
```diff
@@ -257,6 +257,10 @@ maven_install(
...
+        "androidx.arch.core:core-testing:aar:2.1.0",
```
Comment (rdesgroppes) [inline WORKSPACE:263]: I gave a try with these 2 artifacts exactly as written above and indeed faced the issue that `net.bytebuddy:byte-buddy` and `net.bytebuddy:byte-buddy-agent` weren't fetched.
But, despite the theoretical equivalence, stripping out the `jar` packaging type seems to lead to the intended behavior.
```suggestion
        "org.mockito:mockito-core:3.3.3",
        "androidx.arch.core:core-testing:aar:2.1.0",
```
Looking...

Comment (rdesgroppes) [reply WORKSPACE:263]: @artem-zinnatullin I just filed https://github.com/artem-zinnatullin/rules_jvm_external/pull/1 you may merge to your branch or not, at your convenience.

**Comment by rdesgroppes** [comment]
Closes #917.

File: WORKSPACE (hunk 3 above)
Comment (artem-zinnatullin) [reply WORKSPACE:264]: Merged!

File: WORKSPACE (hunk 2 above)
Comment (artem-zinnatullin) [reply WORKSPACE:263]: Yep yep, it's just a convention in our large Android monorepo to see what deps are JVM and what are Android AARs I think it's fine for purpose of this regression test

**Comment by artem-zinnatullin** [comment]
Merged your fix and updated PR title and description for the fix and test to be ready for merge!

File: WORKSPACE (hunk 2 above)
Comment (rdesgroppes) [reply WORKSPACE:263]: Agree.

File: specs.bzl [hunk 4]
```diff
@@ -258,7 +258,7 @@ json = struct(
 #
 def _artifact_to_coord(artifact):
     classifier = (",classifier=" + artifact["classifier"]) if artifact.get("classifier") != None else ""
-    type = (",type=" + artifact["packaging"]) if artifact.get("packaging") != None else ""
+    type = (",type=" + artifact["packaging"]) if artifact.get("packaging") not in (None, "jar") else ""
```
Comment (shs96c) [inline specs.bzl:261]: Nice catch

**Review by shs96c — APPROVED** [review]
LGTM. Thank you for the PR!

**Pull request merged**